docs/ci-cd-gitlab.md

```diff
@@ -0,0 +1,227 @@
+---
+layout: default
+title: Gitlab CI/CD example
+---
+
+# Gitlab CI/CD example
+
+Before discussing the Gitlab CI/CD setup, let's assume you already have a working DNSControl setup. Aren't you there yet? Then first check out the '[Getting Started](https://stackexchange.github.io/dnscontrol/getting-started)' section.
+
+## DNSControl - Demo setup
+
+For this tutorial, there is a [Gitlab repository](https://gitlab.com/cafferata/dnscontrol/) ready with an example DNSControl setup/domain.
+
+This is based on:
+- The domain `cafferata.dev`.
+- The DNS provider [TransIP](https://www.transip.nl/cp/account/api/).
+- The TransIP account `cafferatax`.
+
+For convenience, both configuration files are shown below.
+
+- `dnsconfig.js`
+  ```javascript
+  var PROVIDER_NONE = NewRegistrar('none');
+  var PROVIDER_TRANSIP = NewDnsProvider('transip', '-');
+
+  D('cafferata.dev',
+      PROVIDER_NONE,
+      DnsProvider(PROVIDER_TRANSIP),
+      DefaultTTL('1d'),
+      TXT('spf', [
+          'v=spf1',
+          '-all'
+      ].join(' '))
+  );
+  ```
+- `creds.json`
+  ```json
+  {
+      "transip": {
+          "TYPE": "TRANSIP",
+          "AccountName": "cafferatax",
+          "PrivateKey": "$TRANSIP_PRIVATE_KEY"
+      }
+  }
+  ```
+
+## Gitlab CI - Preparation
+
+You may have noticed that the `creds.json` file contains a variable `$TRANSIP_PRIVATE_KEY`. This variable is populated from the Gitlab CI variables and contain the TransIP API key.
+
+```text
+-----BEGIN PRIVATE KEY-----
+MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQDgJjUQcUrijxul
+NbbN+KD3tDISPDczuJW8HYZRafOc9BgHWuYiBcE+1+9mzHi7FOujuHuTBD84vvRN
+x9buaOLoJNjCvtHDtotYQMhUWF8bnNYyoIIdJGXXcRG0G0+NNreVQiZ+5IQIvhRU
+PXyjvUdQoIaRc9S5NxAMbGLJUHtZ4LEdJKa7HmCXAA5dBYaMMSZjpdBJImSJAtxj
+msseUkc0EiznGgsBVMOkn3VMba2gjo5RDGbfXXLUX3DTOKdwY+hdWktG/gO/3D9l
+hY/gnT/MmXXko3YAcI4eQL8=
+-----END PRIVATE KEY-----
+```
+
+_Example of variable `$TRANSIP_PRIVATE_KEY` contents._
+
+<img src="{{ site.github.url }}/assets/ci-cd-gitlab/settings-ci-cd-variables.png" alt="Gitlab CI/CD variables" style="width: 900px;"/>
+
+<img src="{{ site.github.url }}/assets/ci-cd-gitlab/settings-ci-cd-variables-insert.png" alt="Insert Gitlab CI/CD variable TRANSIP_PRIVATE_KEY" style="width: 900px;"/>
+
+## Gitlab CI - DNSControl preview
+
+Now it's time to apply the power of DNSControl within Gitlab CI merge requests. We'll start by adding the basic Gitlab CI setup. You can view the git diff online in the [Gitlab merge request #1](https://gitlab.com/cafferata/dnscontrol/-/merge_requests/1/diffs). The Gitlab CI setup has also been added for convenience.
+
+`.gitlab-ci.yml`
+
+```yaml
+dnscontrol-preview:
+  stage: 'test'
+  image:
+    name: 'stackexchange/dnscontrol'
+    entrypoint: ['']
+  script:
+    - '/usr/local/bin/dnscontrol version'
+    - '/usr/local/bin/dnscontrol preview'
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "merge_request_event" && $CI_MERGE_REQUEST_IID'
```

Comment:
```suggestion
    - if: '$CI_PIPELINE_SOURCE == "merge_request_event"'
```
`$CI_MERGE_REQUEST_IID` isn't changing anything. It's always set on a MR

Reply:
I agree, removed `$CI_MERGE_REQUEST_IID`.

_https://docs.gitlab.com/ee/ci/jobs/job_control.html#use-predefined-cicd-variables-to-run-jobs-only-in-specific-pipeline-types_